### ISSUE TITLE
test suite is broken on master

### ISSUE THREAD

**Issue by Cerdic** [scssphp/scssphp#246]
on b4d5ecfb

```
1) ScssPhp\ScssPhp\Tests\SassSpecTest::testTests with data set "16/5452. core_functions/color/adjust_color/error/hsl_and_rgb/blue_and_lightness" ('16/5452. core_functions/color...htness', array('', 'a {b: adjust-color(red, $blue...)}\n\n', array(), 'blue_and_lightness'), array('', '', 'Error: RGB parameters may not...et\n\n', array()))
Failed asserting that exception of type "ScssPhp\ScssPhp\Exception\SassException" is thrown.

2) ScssPhp\ScssPhp\Tests\SassSpecTest::testTests with data set "17/5452. core_functions/color/adjust_color/error/hsl_and_rgb/green_and_saturation" ('17/5452. core_functions/color...ration', array('', 'a {b: adjust-color(red, $gree...)}\n\n', array(), 'green_and_saturation'), array('', '', 'Error: RGB parameters may not...et\n\n', array()))
Failed asserting that exception of type "ScssPhp\ScssPhp\Exception\SassException" is thrown.

3) ScssPhp\ScssPhp\Tests\SassSpecTest::testTests with data set "18/5452. core_functions/color/adjust_color/error/hsl_and_rgb/red_and_hue" ('18/5452. core_functions/color...nd_hue', array('', 'a {b: adjust-color(red, $red:...)}\n\n', array(), 'red_and_hue'), array('', '', 'Error: RGB parameters may not...et\n\n', array()))
Failed asserting that exception of type "ScssPhp\ScssPhp\Exception\SassException" is thrown.

4) ScssPhp\ScssPhp\Tests\SassSpecTest::testTests with data set "129/5452. core_functions/color/change_color/error/hsl_and_rgb/blue_and_lightness" ('129/5452. core_functions/colo...htness', array('', 'a {b: change-color(red, $blue...)}\n\n', array(), 'blue_and_lightness'), array('', '', 'Error: RGB parameters may not...et\n\n', array()))
Failed asserting that exception of type "ScssPhp\ScssPhp\Exception\SassException" is thrown.

5) ScssPhp\ScssPhp\Tests\SassSpecTest::testTests with data set "130/5452. core_functions/color/change_color/error/hsl_and_rgb/green_and_saturation" ('130/5452. core_functions/colo...ration', array('', 'a {b: change-color(red, $gree...)}\n\n', array(), 'green_and_saturation'), array('', '', 'Error: RGB parameters may not...et\n\n', array()))
Failed asserting that exception of type "ScssPhp\ScssPhp\Exception\SassException" is thrown.

6) ScssPhp\ScssPhp\Tests\SassSpecTest::testTests with data set "131/5452. core_functions/color/change_color/error/hsl_and_rgb/red_and_hue" ('131/5452. core_functions/colo...nd_hue', array('', 'a {b: change-color(red, $red:...)}\n\n', array(), 'red_and_hue'), array('', '', 'Error: RGB parameters may not...et\n\n', array()))
Failed asserting that exception of type "ScssPhp\ScssPhp\Exception\SassException" is thrown.

7) ScssPhp\ScssPhp\Tests\SassSpecTest::testTests with data set "1042/5452. core_functions/color/scale_color/error/hsl_and_rgb/blue_and_lightness" ('1042/5452. core_functions/col...htness', array('', 'a {b: scale-color(red, $blue:...)}\n\n', array(), 'blue_and_lightness'), array('', '', 'Error: RGB parameters may not...et\n\n', array()))
Failed asserting that exception of type "ScssPhp\ScssPhp\Exception\SassException" is thrown.

8) ScssPhp\ScssPhp\Tests\SassSpecTest::testTests with data set "1043/5452. core_functions/color/scale_color/error/hsl_and_rgb/green_and_saturation" ('1043/5452. core_functions/col...ration', array('', 'a {b: scale-color(red, $green...)}\n\n', array(), 'green_and_saturation'), array('', '', 'Error: RGB parameters may not...et\n\n', array()))
Failed asserting that exception of type "ScssPhp\ScssPhp\Exception\SassException" is thrown.

9) ScssPhp\ScssPhp\Tests\SassSpecTest::testTests with data set "1044/5452. core_functions/color/scale_color/error/hsl_and_rgb/red_and_saturation" ('1044/5452. core_functions/col...ration', array('', 'a {b: scale-color(red, $red: ...)}\n\n', array(), 'red_and_saturation'), array('', '', 'Error: RGB parameters may not...et\n\n', array()))
Failed asserting that exception of type "ScssPhp\ScssPhp\Exception\SassException" is thrown.

10) ScssPhp\ScssPhp\Tests\SassSpecTest::testTests with data set "3063/5452. core_functions/string/to_lower_case/non_ascii" ('3063/5452. core_functions/str..._ascii', array('', '// Only ASCII characters have...)}\n\n', array(), 'non_ascii'), array('@charset "UTF-8";\na {\n  b: ...;\n}\n', '', '', array()))
3063/5452. core_functions/string/to_lower_case/non_ascii
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
 '@charset "UTF-8";\n
 a {\n
-  b: "ÄÇÐØÞ";\n
+  b: "�����";\n
 }'

/Users/cedric/Sites/libs/scssphp/tests/SassSpecTest.php:355

11) ScssPhp\ScssPhp\Tests\SassSpecTest::testTests with data set "4418/5452. libsass/features/extend-selector-pseudoclass" ('4418/5452. libsass/features/e...oclass', array('', 'foo {\n  foo: feature-exists(...n}\n\n', array(), ''), array('foo {\n  foo: true;\n}\n', '', '', array()))
4418/5452. libsass/features/extend-selector-pseudoclass
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
 'foo {\n
-  foo: true;\n
+  foo: false;\n
 }'

/Users/cedric/Sites/libs/scssphp/tests/SassSpecTest.php:355

12) ScssPhp\ScssPhp\Tests\SassSpecTest::testTests with data set "4419/5452. libsass/features/global-variable-shadowing" ('4419/5452. libsass/features/g...dowing', array('', 'foo {\n  foo: feature-exists(...n}\n\n', array(), ''), array('foo {\n  foo: true;\n}\n', '', '', array()))
4419/5452. libsass/features/global-variable-shadowing
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
 'foo {\n
-  foo: true;\n
+  foo: false;\n
 }'

/Users/cedric/Sites/libs/scssphp/tests/SassSpecTest.php:355
```

**Comment by stof** [comment]
that's weird. I don't have any failures locally and CI is green.

Do you have the right version of sass-spec in your vendor folder ? Try running `composer update`

**Comment by Cerdic** [comment]
Good point! I’ll check tomorrow morning

**Comment by Cerdic** [comment]
ok then after a composer update I only have one failing test: 

```
There was 1 failure:

1) ScssPhp\ScssPhp\Tests\SassSpecTest::testTests with data set "3313/5713. core_functions/string/to_lower_case/non_ascii" ('3313/5713. core_functions/str..._ascii', array('', '// Only ASCII characters have...)}\n\n', array(), 'non_ascii'), array('@charset "UTF-8";\na {\n  b: ...;\n}\n', '', '', array()))
3313/5713. core_functions/string/to_lower_case/non_ascii
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
 '@charset "UTF-8";\n
 a {\n
-  b: "ÄÇÐØÞ";\n
+  b: "�����";\n
 }'

/Users/cedric/Sites/libs/scssphp/tests/SassSpecTest.php:355
```

Looks like the  libsass/features/extend-selector-pseudoclass and libsass/features/global-variable-shadowing have been removed from the spec, and the other fails were about the `ScssPhp\ScssPhp\Exception\SassException`

On this failing test the 1.3 version was returning
```
@charset "UTF-8";
a {
  b: "äçðøþ"; }
```

with the same environement, which is also wrong.
Strangely running the full tests was not reporting any error with this version (which looks like to be a second issue itself)

**Comment by Cerdic** [comment]
This is fixed by #250
